components/proxy/src/main/java/com/hotels/styx/spi/ObjectFactories.java

```diff
@@ -53,60 +52,44 @@ private ObjectFactories() {
     }
 
     private static <T> Optional<T> newInstance(Class<T> type, String className, Optional<Path> pathOptional) {
-        Iterable<ClassLoader> classLoaders = pathOptional
-                .map(path -> makeClassLoadersFor(readResourcesFrom(path)))
-                .orElseGet(() -> singleton(ObjectFactories.class.getClassLoader()));
-
-        List<T> instances = newArrayList();
-        for (ClassLoader classLoader : classLoaders) {
-            try {
-                Class<?> aClass = Class.forName(className, false, classLoader);
-                T instance = type.cast(aClass.newInstance());
-                instances.add(instance);
-            } catch (ClassNotFoundException e) {
-                LOGGER.debug(e.getMessage(), e);
-            } catch (InstantiationException | IllegalAccessException e) {
-                LOGGER.error("error loading class={}", className);
-                return Optional.empty();
-            }
-        }
-
-        int size = instances.size();
-        if (size == 0) {
+        ClassLoader classLoader = pathOptional
+                .map(path -> classLoader(readResourcesFrom(path)))
+                .map(ClassLoader.class::cast)
```

Comment:
Why don't you just changed= the return type of the classLoader method to be "ClassLoader" instead of UrlClassLoader and get rid of this cast?